### PR TITLE
[NEW] Voip settings

### PR DESCRIPTION
**Pull request by MartinSchoeler**
Screen Shot 2022-02-17 at 8 47 47 PM
Screen Shot 2022-02-17 at 8 51 29 PM

### CODE REVIEW

**Comment by KevLehman** [comment]
I used github to fix the conflict so I hope i didn't break the linting 🙈

File: client/views/admin/settings/GroupSelector.tsx [hunk 1]
```diff
@@ -7,13 +7,16 @@ import AssetsGroupPage from './groups/AssetsGroupPage';
...
 import OAuthGroupPage from './groups/OAuthGroupPage';
 import TabbedGroupPage from './groups/TabbedGroupPage';
+import VoipGroupPage from './groups/VoipGroupPage';
 
 type GroupSelectorProps = {
 	groupId: GroupId;
 };
 
 const GroupSelector: FunctionComponent<GroupSelectorProps> = ({ groupId }) => {
+	console.log('GroupSelector', groupId);
```
Comment (KevLehman) [inline client/views/admin/settings/GroupSelector.tsx:17]: 👀

File: client/views/admin/settings/GroupSelector.tsx [hunk 2]
```diff
@@ -7,13 +7,16 @@ import AssetsGroupPage from './groups/AssetsGroupPage';
...
 	const group = useSettingStructure(groupId);
+	console.log('group', groupId);
```
Comment (KevLehman) [inline client/views/admin/settings/GroupSelector.tsx:19]: 👀

File: client/views/admin/settings/groups/voip/AssignAgentModal.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+type AssignAgentModalParams = {
+	closeModal: () => void;
+	reload: () => void;
+};
+
+const AssignAgentModal: FC<AssignAgentModalParams> = ({ closeModal, reload }) => {
+	const t = useTranslation();
+	const [agent, setAgent] = useState('');
+	console.log(agent);
```
Comment (KevLehman) [inline client/views/admin/settings/groups/voip/AssignAgentModal.tsx:19]: 👀

File: client/views/admin/settings/groups/voip/AssignAgentModal.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+	const [extension, setExtension] = useState('');
+	const query = useMemo(() => ({ type: 'available' as const, userId: agent }), [agent]);
+
+	const assignAgent = useEndpoint('POST', 'omnichannel/agent/extension');
+
+	const handleAssignment = useMutableCallback(async () => {
+		try {
+			await assignAgent({ userId: agent, extension });
+		} catch (error) {
+			console.log(error);
```
Comment (KevLehman) [inline client/views/admin/settings/groups/voip/AssignAgentModal.tsx:29]: Should we handle this error other way?

File: client/views/admin/settings/groups/voip/RemoveAgentButton.tsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+import { useMutableCallback } from '@rocket.chat/fuselage-hooks';
+import React, { FC } from 'react';
+
+import GenericModal from '../../../../../components/GenericModal';
+import { useSetModal } from '../../../../../contexts/ModalContext';
+import { useEndpoint } from '../../../../../contexts/ServerContext';
+import { useToastMessageDispatch } from '../../../../../contexts/ToastMessagesContext';
+import { useTranslation } from '../../../../../contexts/TranslationContext';
+
+const RemoveTagButton: FC<{ username: string; reload: () => void }> = ({ username, reload }) => {
```
Comment (KevLehman) [inline client/views/admin/settings/groups/voip/RemoveAgentButton.tsx:11]: Should this be RemoveTag or RemoveAgent?

**Review by tiagoevanp — APPROVED** [review]

**Pull request merged**